cpp/collections_module/algorithms/collections.hpp

```diff
@@ -0,0 +1,23 @@
+#include <mgp.hpp>
+
+
+void unionAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory){
+  mgp::memory=memory;
+  const auto arguments=mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try{
+    mgp::List list1=arguments[0].ValueList();
+    mgp::List list2=arguments[1].ValueList();
+    
+    for (size_t i=0; i< list2.Size(); i++){
+        list1.AppendExtend(list2[i]);
```

Comment:
you can use std::move on list2[i] I think, as we are moving values from one list to other, so should be fine. check about move semantics here: https://en.cppreference.com/w/cpp/utility/move